core/src/main/java/io/questdb/griffin/engine/functions/math/WidthBucketFunctionFactory.java

```diff
@@ -0,0 +1,114 @@
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.math;
+
+import org.jetbrains.annotations.Nullable;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.sql.Function;
+import io.questdb.cairo.sql.Record;
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.IntFunction;
+import io.questdb.griffin.engine.functions.QuarternaryFunction;
+import io.questdb.std.IntList;
+import io.questdb.std.Numbers;
+import io.questdb.std.ObjList;
+
+public class WidthBucketFunctionFactory implements FunctionFactory {
+    @Override
+    public String getSignature() {
+        return "width_bucket(DDDI)";
+    }
+
+    @Override
+    public Function newInstance(int position, ObjList<Function> args, IntList argPositions,
+            CairoConfiguration configuration, SqlExecutionContext sqlExecutionContext) throws SqlException {
+        return new WidthBucketFunction(args.getQuick(0), args.getQuick(1), args.getQuick(2), args.getQuick(3));
+    }
+
+    private static class WidthBucketFunction extends IntFunction implements QuarternaryFunction {
+        private final Function operand_func;
+        private final Function low_func;
+        private final Function high_func;
+        private final Function count_func;
+
+        public WidthBucketFunction(Function operand_func, Function low_func, Function high_func, Function count_func) {
+            this.operand_func = operand_func;
+            this.low_func = low_func;
+            this.high_func = high_func;
+            this.count_func = count_func;
+        }
+
+        @Override
+        public Function getLeftEnd() {
+            return operand_func;
+        }
+
+        @Override
+        public Function getCenterLeft() {
+            return low_func;
+        }
+
+        @Override
+        public Function getCenterRight() {
+            return high_func;
+        }
+
+        @Override
+        public Function getRightEnd() {
+            return count_func;
+        }
+
+        @Override
+        public int getInt(Record rec) {
+            double operand = operand_func.getDouble(rec);
+            double low = low_func.getDouble(rec);
+            double high = high_func.getDouble(rec);
+            int count = count_func.getInt(rec);
+
+            if (Double.isNaN(operand) || Double.isNaN(low) || Double.isNaN(high) || count == Numbers.INT_NaN) {
+                return Numbers.INT_NaN;
+            } else if (!(count > 0)) {
+                throw SqlException.$(args, "count must be greater than 0");
```

Comment:
Data accessor methods can't throw `SqlException`. We need to require the checked fields to be a constant or a runtime constant and the check their values in `init()` method. See how it's done in `SplitPartFunctionFactory` as an example.